### DISCUSSION TITLE
V2: how to use new aliases?

### DISCUSSION THREAD

**Discussion by ac130kz** [pydantic/pydantic#6959] [Question]
[This example](https://docs.pydantic.dev/latest/blog/pydantic-v2/#more-powerful-aliases) from the V2 plans fails:
```
pydantic_core._pydantic_core.ValidationError: 1 validation error for Foo
bar
  Field required [type=missing, input_value={'baz': [{'qux': 'a'}, {'...x': 'c'}, {'qux': 'd'}]}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.1/v/missing
```

All I was able to find about aliases is that they are supported now:
* https://github.com/pydantic/pydantic/issues/717
* https://github.com/pydantic/pydantic/pull/3894 
* https://github.com/pydantic/pydantic/discussions/4236
* https://github.com/pydantic/pydantic-core/blob/bcf799f0e8caa9774e839a2d8d30c7f4dd15852f/tests/validators/test_typed_dict.py#L550-L568

But how do I use them then? I want to flatten a model like this:

```json
{
     "module-1": { "<somehash1>": {"property1": 1}, "<somehash2>": {"property1": 2}},
     "module-2": { "<somehash3>": {"property1": 3}, "<somehash4>": {"property1": 4}}
}
```

**Comment by MaciekKrz** [comment]
I'm trying to solve the same problem you have. 
Also when using the `aliases` keyword in `Field` I get the following information:

```PydanticDeprecatedSince20: Using extra keyword arguments on `Field` is deprecated and will be removed. Use `json_schema_extra` instead. (Extra keys: 'aliases')```

Did some play around with 
``` Field(validation_alias=AliasChoices(AliasPath(...)))```
but with no effect.
